### PR TITLE
feat: include argon2, if it is installed

### PR DESCRIPTION
**Pull request by ramsey**

### CODE REVIEW

File: .github/workflows/workflow.yml
Comment (ramsey) [inline .github/workflows/workflow.yml]: I don't think this will have any effect until it's merged into the master branch.

**Review by Skatox — APPROVED** [review]

File: .github/workflows/workflow.yml
Comment (ahawkins) [reply .github/workflows/workflow.yml]: I think adding this broke CI checks on PRs. There are no action runs associated with this PR. What's the reasoning behind this change?

File: .github/workflows/workflow.yml
Comment (ahawkins) [inline .github/workflows/workflow.yml]: Nice. Did you notice a version upgrade then just try it?

**Review by ahawkins — APPROVED** [review]
I have no problem adding an optional dependency like this. Are we using `argon2` in some way?

Approved, assuming we get CI running again.

File: .github/workflows/workflow.yml
Comment (ramsey) [reply .github/workflows/workflow.yml]: I added this because the workflow didn't run. Then, after I added it, the workflow still didn't run.

File: .github/workflows/workflow.yml
Comment (ramsey) [reply .github/workflows/workflow.yml]: I saw the version number so just checked to see if there was a newer version, and there is.

**Comment by ramsey** [comment]
> Are we using argon2 in some way?

Yes. We're using it in production, and we have some tests that rely on it being active. If I run these tests locally (outside the containers) they fail. Adding this lets me run all the unit tests without needing to run them in the container.

File: bin/install
Comment (ahawkins) [inline bin/install:36]: Is this necessary? I'm not sure what role `pkg-config` plays with this dependency. Is `--with-argon2=/foo` enough?

Comment (ramsey) [reply bin/install:36]: [The PHP docs say:](https://www.php.net/manual/en/password.installation.php)

> Prior to PHP 8.1.0, it was possible to specify the argon2 directory with `--with-password-argon2[=DIR]`.

So, I asked about it: https://chat.stackoverflow.com/transcript/message/56607367#56607367

```
ramsey: The password hashing docs say, "Prior to PHP 8.1.0, it was possible to specify the argon2 directory with --with-password-argon2[=DIR]." So, does that mean libargon2 is bundled with php-src now? php.net/manual/en/password.installation.php
Derick: No, you just can't specify the directory.
Derick: The "new" way depends on pkg-config and it supplying the right directory /cc @ramsey
ramsey: gotcha
Derick: (I didn't know that, I had to research it)
```

Most everything is moving away from (or has already moved away from) specifying the directory and using `pkgconfig` instead.

Comment (ramsey) [reply bin/install:36]: The `configure` script uses `pkg-config` like the following to get the linker and C flags for a package, which will then go into the generated `Makefile` and passed along to the compiler when PHP is built.

A command like this:

``` shell
PKG_CONFIG_PATH="$(brew --prefix argon2)/lib/pkgconfig" \
    pkg-config --libs --cflags libargon2
```

Will return something like this:

```
-I/opt/homebrew/Cellar/argon2/20190702_1/include -L/opt/homebrew/Cellar/argon2/20190702_1/lib -largon2
```

Comment (ahawkins) [reply bin/install:36]: Wow. Great information. Thank you!

**Review by ahawkins — APPROVED** [review]

**Pull request merged**
